google-cloud-firestore/lib/google/cloud/firestore/query.rb

```diff
@@ -0,0 +1,652 @@
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          field_refs = fields.flatten.compact.map do |field|
+            field = FieldPath.parse field unless field.is_a? FieldPath
+            StructuredQuery::FieldReference.new \
+              field_path: field.formatted_string
+          end
+
+          new_query.select ||= StructuredQuery::Projection.new
+          field_refs.each do |field_ref|
+            new_query.select.fields << field_ref
+          end
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # @private This is marked private can't be removed.
+        #
+        # Selects documents from all collections, immediate children and nested,
+        # of where the query was created from.
+        #
+        # @return [Query] New query with `all_descendants` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.all_descendants
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def all_descendants
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          if new_query.from.empty?
+            fail "missing collection_id to specify descendants."
+          end
+
+          new_query.from.last.all_descendants = true
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # @private This is marked private can't be removed.
+        #
+        # Selects only documents from collections that are immediate children of
+        # where the query was created from.
+        #
+        # @return [Query] New query with `direct_descendants` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.direct_descendants
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def direct_descendants
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          if new_query.from.empty?
+            fail "missing collection_id to specify descendants."
+          end
+
+          new_query.from.last.all_descendants = false
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # Filters the query on a field.
+        #
+        # @param [String, Symbol] field A field mask to filter results with.
```

Comment:
This comment seems out-of-place here.

Reply:
Yep. It wasn't removed when the new argument text was added. Removed.